In [1]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/happy-880/IT326-Project-/refs/heads/main/Dataset/Raw_dataset.csv' , sep=';')
df.columns = df.columns.str.strip()
pd.set_option('display.max_columns', None)

# IT326 Data Mining Project

## Phase 1

### 1. Project Goal
The goal of this project is to apply data mining techniques to analyze student data and identify patterns related to students academic outcomes.

**Classification:**

The classification task aims to classify students into three categories:

• Dropout
• Enrolled
• Graduate

**Clustering:**

The clustering task aims to group students based on similarities in their characteristics and identify meaningful groups within the dataset.

### 2. Dataset Source

 **Dataset Name:** Predict Students' Dropout and Academic Success

 **Source Repository:** UCI Machine Learning Repository

 **URL:** https://archive.ics.uci.edu/dataset/697/predict+students+dropout+and+academic+success


### 3. Dataset Description
This dataset comprises higher education student records, combining demographic background, socioeconomic indicators, and academic performance data. The primary objective is to evaluate student academic trajectories and classify them into three groups: Dropout, Enrolled, or Graduate. This allows for early detection of students who may require academic support to complete their studies successfully.

#### 3.1 Number of Attributes
**Number of Attributes:** 36
attributes and 1 class attribute (Target)



#### 3.2 Data Types of Attributes
**Types:** Real, Categorical, and Integer



In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 37 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   int64  
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4424 non-null   int64  
 4   Daytime/evening attendance                      4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Previous qualification (grade)                  4424 non-null   float64
 7   Nacionality                                     4424 non-null   int64  
 8   Mother's qualification                          4424 non-null   int64  
 9   Father's qualification                   

#### 3.3 Number of Records
**The dataset contains** 4,424 records, where each record represents one student.

#### 3.4 Class Attribute
**Target**



#### 3.5 Number of Instances for Each Class
**Graduate:** 2,209,
**Dropout:** 1,421,
**Enrolled:** 794.


In [3]:
print(df['Target'].value_counts())

Target
Graduate    2209
Dropout     1421
Enrolled     794
Name: count, dtype: int64


#### 3.6 Sample of Raw Dataset
The following table shows a sample of the raw dataset, including the first five records and all available attributes.



In [4]:
df.head()

,Marital status,Application mode,Application order,Course,Daytime/evening attendance,Previous qualification,Previous qualification (grade),Nacionality,Mother's qualification,Father's qualification,Mother's occupation,Father's occupation,Admission grade,Displaced,Educational special needs,Debtor,Tuition fees up to date,Gender,Scholarship holder,Age at enrollment,International,Curricular units 1st sem (credited),Curricular units 1st sem (enrolled),Curricular units 1st sem (evaluations),Curricular units 1st sem (approved),Curricular units 1st sem (grade),Curricular units 1st sem (without evaluations),Curricular units 2nd sem (credited),Curricular units 2nd sem (enrolled),Curricular units 2nd sem (evaluations),Curricular units 2nd sem (approved),Curricular units 2nd sem (grade),Curricular units 2nd sem (without evaluations),Unemployment rate,Inflation rate,GDP,Target
0,1,17,5,171,1,1,122.0,1,19,12,5,9,127.3,1,0,0,1,1,0,20,0,0,0,0,0,0.000000,0,0,0,0,0,0.000000,0,10.8,1.4,1.74,Dropout
1,1,15,1,9254,1,1,160.0,1,1,3,3,3,142.5,1,0,0,0,1,0,19,0,0,6,6,6,14.000000,0,0,6,6,6,13.666667,0,13.9,-0.3,0.79,Graduate
2,1,1,5,9070,1,1,122.0,1,37,37,9,9,124.8,1,0,0,0,1,0,19,0,0,6,0,0,0.000000,0,0,6,0,0,0.000000,0,10.8,1.4,1.74,Dropout
3,1,17,2,9773,1,1,122.0,1,38,37,5,3,119.6,1,0,0,1,0,0,20,0,0,6,8,6,13.428571,0,0,6,10,5,12.400000,0,9.4,-0.8,-3.12,Graduate
4,2,39,1,8014,0,1,100.0,1,37,38,9,9,141.5,0,0,0,1,0,0,45,0,0,6,9,5,12.333333,0,0,6,6,6,13.000000,0,13.9,-0.3,0.79,Graduate
